eos-enable-zram

```diff
@@ -13,13 +13,79 @@ if [ "$1" -eq 0 ]; then
     exit 0
 fi
 
-ram_size_kb=$(awk '/MemTotal/{print $2}' /proc/meminfo)
+# List and disable any active swap areas
+active_swaps=$(swapon --show=NAME --raw --noheadings)
+echo "Active swap areas before $(basename $0): ${active_swaps}"
+swapoff -a
 
+# Enable zram
+ram_size_kb=$(awk '/MemTotal/{print $2}' /proc/meminfo)
 if [ ! -e /sys/block/zram0 ]; then
     modprobe zram
 fi
 echo $(($ram_size_kb * 2))K > /sys/block/zram0/disksize
-# For now, disable the swap partition if in use
-swapoff -a
 mkswap /dev/zram0
 swapon /dev/zram0
+
+if [ -z "${active_swaps}" ]; then
+    exit 0
+fi
+
+# Find root partition, root partition number, root disk, and table type
+orig_root_part=$(systemctl show -p What sysroot.mount)
```

Comment:
This all looks very familiar – should we extract this code into a function/separate program rather than copy-pasting it?

Reply:
Yes, this all comes from `endless-repartition.sh` which runs from the initrd, so we would need to deploy it twice anyway. While we could factor it on a separate file, I'm not sure the benefit is greater from the effort to do it. Maybe we can do it as a separate task, in the all-too-familiar technical-debt fashion? :-)

Reply:
Yeah, I don't think it's worth it in that case. Maybe in a different world these would be C programs that could share an internal static library.